src/tools/rbd_mirror/image_replayer/snapshot/Replayer.cc

```diff
@@ -1500,6 +1548,38 @@ bool Replayer<I>::is_replay_interrupted(std::unique_lock<ceph::mutex>* locker) {
   return false;
 }
 
+template <typename I>
+void Replayer<I>::register_perf_counters() {
+  dout(5) << dendl;
+
+  ceph_assert(ceph_mutex_is_locked_by_me(m_lock));
+  ceph_assert(m_perf_counters == nullptr);
+
+  auto cct = static_cast<CephContext *>(m_state_builder->local_image_ctx->cct);
+  auto prio = cct->_conf.get_val<int64_t>("rbd_mirror_image_perf_stats_prio");
+  PerfCountersBuilder plb(g_ceph_context, "rbd_mirror_image_" + m_image_spec,
+                          l_rbd_mirror_first, l_rbd_mirror_replay_latency);
```

Comment:
I didn't include the latency for now because I don't know what kind of latency we want? Do we want latency per object copied or "latency" for a whole sync or both or no latency metrics at all?

If  we choose both what should we report back to the global metrics for the whole daemon (that would be collocated with the journal replayer basically)?

Reply:
I think the latency should correspond to what you count. If it is objects (as in your current implementation) then it should be object copy latency. If it is snapshots (as I suggest) then it should be a snapshot copy latency.

If we decide to use different counter names for snapshot based mirroring (as I suggested in another comment), then instead of "latency" we could use "time" here, which seems to have more sense.

Reply:
I added "snapshots_time", not sure if that's ideal or not though